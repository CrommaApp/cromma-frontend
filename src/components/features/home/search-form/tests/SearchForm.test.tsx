import React from 'react';
import { render, screen } from '@testing-library/react';
import SearchForm from '@components/features/home/search-form';
import SearchNewsService from '@services/search/search-news';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

describe('SearchForm', () => {
	let searchNewsService: SearchNewsService;
	let moveToReultPage: (keyword: string) => void;

	const clickSearchButton = () => {
		const searchButton = screen.getByRole('button', {
			name: 'Search',
		});

		userEvent.click(searchButton);
	};

	beforeEach(() => {
		searchNewsService = new SearchNewsService();
		moveToReultPage = jest.fn();

		render(<SearchForm searchNewsService={searchNewsService} moveToReultPage={moveToReultPage} />);
	});

	it('Snapshot', () => {
		const component = renderer.create(
			<SearchForm searchNewsService={searchNewsService} moveToReultPage={moveToReultPage} />,
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Go to the result page when success to search', async () => {
		const searchInput = screen.getByRole('searchbox', {
			name: 'keywords',
		});

		userEvent.type(searchInput, 'korea');

		clickSearchButton();

		expect(moveToReultPage).toHaveBeenCalledWith('korea');
	});

	it('Keyword must be entered', async () => {
		clickSearchButton();

		expect(moveToReultPage).not.toBeCalled();
	});
});
