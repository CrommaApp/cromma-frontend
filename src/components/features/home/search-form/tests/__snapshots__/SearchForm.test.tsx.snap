// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm Snapshot 1`] = `
<form
  aria-labelledby="search_guide"
  className="sc-bdvvtL bQPxZi"
  onSubmit={[Function]}
>
  <label
    className="a11y-hidden"
    htmlFor="search_input"
  >
    keywords
  </label>
  <input
    aria-required="true"
    id="search_input"
    onChange={[Function]}
    type="search"
    value=""
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
